src/main/java/com/microsoft/sqlserver/jdbc/SQLServerResultSet.java

```diff
@@ -389,6 +389,8 @@ boolean onDone(TDSReader tdsReader) throws SQLServerException {
                     SQLServerException.makeFromDriverError(stmt.connection, stmt, form.format(msgArgs), null, false);
                 }
 
+                // decrementUnprocessedResponseCount() is not necessary here. It will over decrement if added.
```

Comment:
I agree that we would not want to blindly decrement here. But would this onDone for ClientCursorInitializer ever get called in the context of a final done token? Or do other onDone methods always handle the final done tokens? (I admit, I don't even know where ClientCursorInitializer is used.)

Reply:
Yes, it does get called when there's an error for a final done. My mistake, it looks like I added a decrement in the other onDone in SQLServerResultSet, but not here when I was cleaning up my PR in the end to push.

The decrement is not necessary outside the "if" as I found that from debugging the tests it would over decrement. The decrement is accounted for in the other onDone when there is no error.